tests/JsonReaderTests.cs

```diff
@@ -210,6 +210,52 @@ public void DateTime_With_Invalid_Input(string expectedErrorToken, string json)
                          expectedErrorToken);
     }
 
+    [Fact]
+    public void DateTimeOffset_Moves_Reader()
+    {
+        TestMovesReaderPastReadValue(JsonReader.DateTimeOffset(), "'2022-02-02T12:34:56-01:00'");
+    }
+
+    [Theory]
+    [InlineData(2022, 2, 2, 0, 0, 0, 0, null, null, "'2022-02-02'")]
+    [InlineData(2022, 2, 2, 12, 34, 0, 0, null, null, "'2022-02-02T12:34'")]
+    [InlineData(2022, 2, 2, 12, 34, 0, 0, 0, 0, "'2022-02-02T12:34Z'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 0, null, null, "'2022-02-02T12:34:56'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 0, 0, 0, "'2022-02-02T12:34:56Z'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 78, null, null, "'2022-02-02T12:34:56.078'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 78, 0, 0, "'2022-02-02T12:34:56.078Z'")]
+    [InlineData(2022, 2, 2, 12, 34, 0, 0, 1, 0, "'2022-02-02T12:34:00+01:00'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 0, 2, 0, "'2022-02-02T12:34:56+02:00'")]
+    [InlineData(2022, 2, 2, 12, 34, 56, 78, 0, -5, "'2022-02-02T12:34:56.078-00:05'")]
+    public void DateTimeOffset_With_Valid_Input(int year, int month, int day, int hour, int minute, int second, int millisecond, int? hourOffset, int? minuteOffset, string json)
```

Comment:
nit: wrap long parameters list in groups:

```suggestion
    public void DateTimeOffset_With_Valid_Input(int year, int month, int day, int hour, int minute, int second, int millisecond,
                                                int? hourOffset, int? minuteOffset,
                                                string json)
```